cli/cmd/check.go

```diff
@@ -0,0 +1,182 @@
+			WithHintAnchor("l5d-smi-pods-injection").
+			Warning().
+			WithCheck(func(ctx context.Context) error {
+				// Check if SMI Extension pods have been injected
+				pods, err := hc.KubeAPIClient().GetPodsByNamespace(ctx, smiNamespace)
+				if err != nil {
+					return err
+				}
+				return healthcheck.CheckIfDataPlanePodsExist(pods)
+			}))
+
+	checkers = append(checkers,
+		*healthcheck.NewChecker("SMI extension pods are running").
+			WithHintAnchor("l5d-smi-pods-running").
+			Fatal().
+			WithRetryDeadline(hc.RetryDeadline).
+			SurfaceErrorOnRetry().
+			WithCheck(func(ctx context.Context) error {
+				pods, err := hc.KubeAPIClient().GetPodsByNamespace(ctx, smiNamespace)
+				if err != nil {
+					return err
+				}
+
+				// Check for relevant pods to be present
+				err = healthcheck.CheckForPods(pods, []string{"smi-adaptor"})
+				if err != nil {
+					return err
+				}
+
+				return healthcheck.CheckPodsRunning(pods, "")
+
+			}))
+
+	checkers = append(checkers,
+		*healthcheck.NewChecker("SMI extension proxies are healthy").
+			WithHintAnchor("l5d-smi-proxy-healthy").
+			Fatal().
+			WithRetryDeadline(hc.RetryDeadline).
+			SurfaceErrorOnRetry().
+			WithCheck(func(ctx context.Context) error {
+				return hc.CheckProxyHealth(ctx, hc.ControlPlaneNamespace, smiNamespace)
+			}))
+
+	return healthcheck.NewCategory(linkerdSMIExtensionCheck, checkers, true)
+}
+
+func newCheckOptions() *checkOptions {
+	return &checkOptions{
+		wait:   300 * time.Second,
+		output: healthcheck.TableOutput,
+	}
+}
+
+func (options *checkOptions) validate() error {
+	if options.output != healthcheck.TableOutput && options.output != healthcheck.JSONOutput {
+		return fmt.Errorf("Invalid output type '%s'. Supported output types are: %s, %s", options.output, healthcheck.JSONOutput, healthcheck.TableOutput)
+	}
+	return nil
+}
+
+// newCmdCheck generates a new cobra command for the SMI extension.
+func newCmdCheck() *cobra.Command {
+	options := newCheckOptions()
+	cmd := &cobra.Command{
+		Use:   "check [flags]",
+		Args:  cobra.NoArgs,
+		Short: "Check the SMI extension for potential problems",
+		Long: `Check the SMI extension for potential problems.
+
+The check command will perform a series of checks to validate that the SMI
+extension is configured correctly. If the command encounters a failure it will
+print additional information about the failure and exit with a non-zero exit
+code.`,
+		Example: `  # Check that the SMI extension is up and running
+  linkerd smi check`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return configureAndRunChecks(stdout, stderr, options)
+		},
+	}
+
+	cmd.Flags().StringVarP(&options.output, "output", "o", options.output, "Output format. One of: basic, json")
+	cmd.Flags().DurationVar(&options.wait, "wait", options.wait, "Maximum allowed time for all tests to pass")
+	cmd.Flags().BoolVar(&options.proxy, "proxy", options.proxy, "Also run data-plane checks, to determine if the data plane is healthy")
```

Comment:
This flag doesn't do anything does it? Is it required when calling through `linkerd check smi' ? If so, can it be hidden in this check's help?

Reply:
Yep. This is required when `linkerd check --proxy` is ran with SMI installed, as those same flags are passed, and the cmd would fail if the flag is not supported.

👍🏼 on marking them as hidden!